pennylane/compiler/python_compiler/transforms/convert_to_mbqc_formalism.py

```diff
@@ -0,0 +1,770 @@
+        Args:
+            qb_in_reg (QubitType): A qubit in the global qubit register.
+            aux_res_qubit (QubitType): The result auxiliary qubit.
+            op (CustomOp) : A gate operatio object.
+            rewriter (pattern_rewriter.PatternRewriter): A pattern rewriter.
+
+        Return:
+            The result qubits, which are the swapping result of the qubit in the global register and the auxiliary result qubit.
+        """
+        # NOTE: IdentityOp inserted here is a temporal solution to fix the issue that SWAPOp can't accept
+        # `res_aux_qubit` as an in_qubits variable, i.e. SWAPOp = CustomOp(in_qubits=(target_qubit, res_aux_qubit), gate_name="SWAP").
+        # The error message would be "| Error while applying pattern: 'NoneType' object has no attribute 'add_use'". Is it a upstream
+        # issue or caused by the way how we define the `CustomOp` operation? Not sure why.
+        # NOTE: It is also worth noting here. The target qubit in the global register is measured and I assume that
+        # physically the target qubit can not be swapped with the result auxiliary qubit, right?
+        # TODOS: We need more clarifications for the questions above in the following steps.
+        IdentityOp = CustomOp(in_qubits=(aux_res_qubit), gate_name="Identity")
+        rewriter.insert_op(IdentityOp, InsertPoint.before(op))
+
+        in_qubits = (qb_in_reg, IdentityOp.results[0])
+
+        SWAPOp = CustomOp(in_qubits=in_qubits, gate_name="SWAP")
+        rewriter.insert_op(SWAPOp, InsertPoint.before(op))
+
+        return SWAPOp.results
+
+    def _deallocate_aux_qubits(
+        self,
+        graph_qubits_dict: dict,
+        qb_in_reg_key: list[int],
+        op: CustomOp,
+        rewriter: pattern_rewriter.PatternRewriter,
+    ):
+        """Deallocate the auxiliary qubits in the graph qubit dict.
+        Args:
+            graph_qubits_dict (dict) : A dict stores all qubits in a graph state.
+            qb_in_reg_key (list[int]) : A list of keys to qubits in the global register.
+            op (CustomOp) : A gate operatio object.
+            rewriter (pattern_rewriter.PatternRewriter): A pattern rewriter.
+        """
+        # Deallocate aux_qubits
+        for node in graph_qubits_dict:
+            if node not in qb_in_reg_key:
+                deallocQubitOp = DeallocQubitOp(graph_qubits_dict[node])
+                rewriter.insert_op(deallocQubitOp, InsertPoint.before(op))
+
+    # pylint: disable=no-self-use
+    @pattern_rewriter.op_type_rewrite_pattern
+    def match_and_rewrite(
+        self, root: func.FuncOp | IfOp | WhileOp | ForOp, rewriter: pattern_rewriter.PatternRewriter
+    ):  # pylint: disable=arguments-differ, cell-var-from-loop
+        """Match and rewrite for converting to the MBQC formalism."""
+
+        for region in root.regions:
+            for op in region.ops:
+                if isinstance(op, CustomOp) and op.gate_name.data in [
```

Comment:
Nothing to be done yet, but it may be useful to migrate the if/else body to functions instead.
We can come back to this.